### ISSUE TITLE
Document deployment to Seldon Core

### ISSUE THREAD

**Issue by axsaucedo** [SeldonIO/MLServer#237]

**Comment by divyadilip91** [comment]
Hi @adriangonz 
Is there a document available on how to serve custom model using ML server and deploy it on Seldon core.
Awaiting your response

**Comment by adriangonz** [comment]
Hey @divyadilip91, there isn't anything documenting the full end-to-end process **yet**. However, it should be possible to put something together from these sources:

- Writing a custom inference runtime for MLServer: https://mlserver.readthedocs.io/en/latest/examples/custom/README.html
- Current MLServer's `Dockerfile`: https://github.com/SeldonIO/MLServer/blob/master/Dockerfile
- Configuring a custom inference server in Seldon Core: https://docs.seldon.io/projects/seldon-core/en/latest/servers/custom.html

Let me know if you have any questions! :+1:

**Comment by divyadilip91** [comment]
Hi @adriangonz 
I am a bit confused here with the above links. How does the first point & third point differ and how would the dockerfile of current MLserver be used? It would be helpful if you could explain a bit more.

My usecase is I want to develop a microservice based on fastapi framework on seldon core and I learned that MLserver is based on Fast API framework. So how do I proceed further to implementation?

**Comment by adriangonz** [comment]
Hey @divyadilip91, the steps to serve a custom Seldon Core model with MLServer would follow something along these lines:

1. Write your custom inference runtime for MLServer. For this, you can follow the ["Writing a custom inference runtime"](https://mlserver.readthedocs.io/en/latest/examples/custom/README.html) example from the MLServer docs (i.e. the 1st link).
2. Once you have your custom code, package it up into a custom Docker image. You can base this image on the existing `seldonio/mlserver` one (you can find the [Dockerfile](https://github.com/SeldonIO/MLServer/blob/master/Dockerfile) on the 2nd link above). The simplest (but not necessarily the most efficient) would be something along these lines:
   ```
   FROM seldonio/mlserver
   
   # Enable our custom runtime by default
   ENV MLSERVER_MODEL_IMPLEMENTATION="my_custom_module.MyCustomInferenceRuntime"
   
   # Install any extra required deps
   COPY requirements.txt .
   RUN pip install -r requirements.txt
   
   # Copy our custom so that it's accessible on the image
   COPY my_custom_module.py my_custom_module.py
   ```
3. (Optional) Configure a custom inference server in Seldon Core, so that you can serve your model through the `implementation` key of the `SeldonDeployment` CRD. This can be done by adding a new key under `predictor_servers` pointing to the custom you've built in step 2. The process for this can be found in the [Seldon Core docs](https://docs.seldon.io/projects/seldon-core/en/latest/servers/custom.html) (i.e. the 3rd link).

On the protocol point, currently MLServer only supports the V2 (or `kfserving`) protocol, so out-of-the-box it's not possible to serve models through the `seldon` or other (e.g. `tf`) protocols. Seldon Core fully supports the V2 protocol though. You can find details on how to query V2 models in the [Seldon Core docs](https://docs.seldon.io/projects/seldon-core/en/latest/graph/protocols.html#v2-kfserving-protocol). 

It's also possible to add custom protocols to your inference runtime as well. This can be leveraged to add support for the `seldon` protocol on your custom runtime. This is not covered on the docs yet, but happy to provide more info on this if you need.

Let me know if that answers your questions!

**Comment by adriangonz** [comment]
Fixed by #389 

Deployment guide can now be seen in the docs: 

https://mlserver.readthedocs.io/en/latest/user-guide/deployment/seldon-core.html
